### ISSUE TITLE
Redis sessions not working

### ISSUE THREAD

**Issue by Boycce** [balderdashy/sails#3589]
Hey, i'm not sure if this issue is related to #3586 
I followed your [guide](http://sailsjs.org/documentation/reference/configuration/sails-config-session) for setting up a more appropriate session store using redis, but cannot seem to set any cookies. I am using `sails 0.11.5` with `connect-redis 1.4.5` and have this in my config:

```
  adapter: 'connect-redis',
  host: 'localhost',
  port: 6379,
  db: 0,
```

What might i be missing here? Also I feel Sails should default to using redis for sessions as i was getting a *140ms* delay on each response from the native memory session store on my server. This bottleneck was also quite elusive.

**Comment by mikermcneil** [comment]
@Boycce will try to reproduce asap-- offhand, I don't see any problems with your usage, but I'll need to take a closer look. Would you mind adding your npm and node version to this issue for posterity in the mean time?

**Comment by Boycce** [comment]
Sweet sure, `node 4.1.0`, `npm 2.14.3`. Im quite new to redis, I don't need to start it do I?

**Comment by mikermcneil** [comment]
@Boycce thanks for posting the version info!

> Im quite new to redis, I don't need to start it when we're dealing with connect-redis do I?

Hmm, sounds like that might be the culprit.  You'll want to open a new terminal window and run `redis-server` (that starts your redis db locally so that connect-redis can connect to it).  If memory serves, [part 2 of this video](https://courses.platzi.com/courses/develop-apps-sails-js/) touches on this (can't remember if I did local redis as well as redis on heroku or not offhand though)

**Comment by udanpe** [comment]
When redis server not running. Sails (with passport) don't show any error.
npm 3.3.12; node 5.4.0; passport 0.3.2; passport-local: 1.0.0
brew services stop redis.
```
passport.use(new LocalStrategy(), function (email, password, done) {
    User.findOne({email: email}, function (err, user) {
      if (err) return done(err);
      if (!user) return done(null, false, {message: 'Incorrect email.'});
      bcrypt.compare(password, user.password, function (err, res) {
        if (!res) return done(null, false, { message: 'Invalid Password'});
        return done(null, user, {message: 'Logged In Successfully'});
      });
    });
  }
));
```

**Comment by irlnathan** [comment]
Howdy @Boycce,

After reading the thread it sounds like y'all might have discovered the issue – wasn't running the redis server locally, but I decided to create a quick project and see what errors I received.

My first thought was there's no need for
```
adapter: 'connect-redis',
host: 'localhost',
port: 6379,
db: 0,
```
because its optional and using the default setup.

After creating a new project sails v0.11.5 and uncommenting `redis` in session.js, I started sails via `sails lift`.
> FYI: No adapter installed and redis not running locally yet.
I get the helpful messages

```
Do you have the Connect session adapter installed in this project?
Try running the following command in your project's root directory:
npm install connect-redis@1.4.5
(Note that connect-redis@1.5.0 introduced breaking changes- make sure you have v1.4.5 installed!)
```

So far so good.  Next I installed the adapter `connect-redis@1.4.5`. and then attempted to start the Sails server using `sails lift`. The Sails server started without an error. 

So despite you being able to solve this issue, an explicit error message would be helpful here if the adapter fails to make a connection to a redis server.
